src/app/shared/services/product.service.ts

```diff
@@ -92,23 +92,24 @@ export class ProductService {
   }
 
   private filterProducts() {
-    this.filteredProducts = [];
     if (this.filterInput) {
+      this.filteredProducts = [];
+
       this.filterInput.forEach(item => {
 
         switch (item.name) {
           case 'نوع' :
-            this.filteredProducts.concat(this.products.filter(product => item.values.includes(product.product_type.name)));
+            this.filteredProducts = this.filteredProducts.concat(this.products.filter(product => item.values.includes(product.product_type.name)));
```

Comment:
check filtered products not to be recurrent, some products may add several times to array, look this problem in other switch cases too